src/pages/AddProduct.js

```diff
@@ -1,68 +1,54 @@
 import React, { useState } from 'react';
 import axios from 'axios';
 import './styles.css';
+import { useProductsContext } from "../context/products_context";
 
-const AddProduct = () => {
-  const [name, setName] = useState('');
-  const [price, setPrice] = useState('');
-  const [description, setDescription] = useState('');
-  const [image, setImage] = useState('');
-  const [category, setCategory] = useState('');
-  const [company, setCompany] = useState('');
+const emptyProduct = {
+  name: '',
+  price: '',
+  description: '',
+  image: '',
+  imageFile: undefined,
+  category: '',
+  company: '',
+  featured: false,
+  freeShipping: false,
+  inventory: 15
+}
 
-  const [featured, setFeatured] = useState(false);
-  const [freeShipping, setFreeShipping] = useState(false);
-  const [inventory, setInventory] = useState(15);
+const AddProduct = () => {
+  const { addProduct } = useProductsContext();
+  const [productData, setProductData] = useState(emptyProduct);
   const [message, setMessage] = useState(''); //message 'product'added'
 
   const handleSubmit = async (e) => {
-    e.preventDefault();
-    const formData = new FormData(); // instantiate a 'form-data' object in the browser to upload the image file
-    formData.set('image', image);
-    const imageResponse = await axios.post(
-      'https://ecommerce-6kwa.onrender.com/api/v1/products/uploadImage',
-      formData,
-      {
-        withCredentials: true,
-        headers: { 'Content-Type': 'multipart/form-data' },
-      }
-    );
-    console.dir(imageResponse);
-
-    const product = {
-      name,
-      price,
-      description,
-      image: imageResponse.data.image,
-      category,
-      company,
-
-      featured,
-      freeShipping,
-      inventory,
-    };
-
     try {
-      const response = await axios.post(
-        'https://ecommerce-6kwa.onrender.com/api/v1/products',
-        product,
-        { withCredentials: true }
+      e.preventDefault();
+      setMessage('');
+      const formData = new FormData(); // instantiate a 'form-data' object in the browser to upload the image file
+      formData.set('image', productData.imageFile);
+      const imageResponse = await axios.post(
+        'https://ecommerce-6kwa.onrender.com/api/v1/products/uploadImage',
+        formData,
+        {
+          withCredentials: true,
+          headers: { 'Content-Type': 'multipart/form-data' },
+        }
       );
-      console.log('Product added successfully:', response.data);
-      setMessage(`Product added successfully`);
+      console.dir(imageResponse);
 
-      // Reset form fields
-      setName('');
-      setPrice('');
-      setDescription('');
-      setImage('');
-      setCategory('');
-      setCompany('');
-      setFeatured(false);
-      setFreeShipping(false);
-      setInventory(15);
```

Comment:
Having separate state setters for each field in an object, makes React do a little more work. As well, it's easy to forget about a single field or accidentally set it to an incorrect value when writing the code. Instead, my suggestion is to just group related things together in an object (in this case, as `setProductData(<product object>)` altogether).